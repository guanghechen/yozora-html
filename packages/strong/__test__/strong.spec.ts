import type { Strong as IStrong } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderStrong from '../src'

describe('snapshot', function () {
  it('basic-1', function () {
    const node = {
      type: 'strong',
      children: [
        {
          type: 'text',
          value: 'yozora is cool!',
        },
      ],
    }
    expect(renderStrong(node as IStrong, renderChildren)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const node = {
      type: 'strong',
      children: [
        {
          type: 'emphasis',
          children: [
            {
              type: 'text',
              value: 'Hello',
            },
          ],
        },
        {
          type: 'text',
          value: ', ',
        },
        {
          type: 'strong',
          children: [
            {
              type: 'text',
              value: 'World',
            },
          ],
        },
      ],
    }
    expect(renderStrong(node as IStrong, renderChildren)).toMatchSnapshot()
  })
})
