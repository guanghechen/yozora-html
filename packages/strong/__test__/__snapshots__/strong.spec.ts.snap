// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic-1 1`] = `"<strong class="yozora-strong"><span class="yozora-text">yozora is cool!</span></strong>"`;

exports[`snapshot basic-2 1`] = `"<strong class="yozora-strong"><em class="yozora-emphasis"><span class="yozora-text">Hello</span></em><span class="yozora-text">, </span><strong class="yozora-strong"><span class="yozora-text">World</span></strong></strong>"`;
